File: CustomComponents/MyFloatingButton/app/src/main/java/com/wparja/myfloatingbutton/MainActivity.java
package com.wparja.myfloatingbutton;

import androidx.annotation.RequiresApi;
import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.content.ContentValues;
import android.database.Cursor;
import android.graphics.Color;
import android.net.Uri;
import android.os.Build;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.FrameLayout;
import android.widget.LinearLayout;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;

public class MainActivity extends AppCompatActivity {

    private static final String INTERFACE_ID = "interface_id";
    private static final String OEM_ID = "oem_id";
    private static final String DATE = "date";
    private static final String SENSOR_VALUE = "value";

    @SuppressLint("ResourceAsColor")
    @RequiresApi(api = Build.VERSION_CODES.M)
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final LinearLayout l = findViewById(R.id.main);
        Button b = new Button(this);

        b.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                for (int i = 0; i < l.getChildCount(); i++) {
                    l.removeView(l.getChildAt(i--));
                }
            }
        });
        b.setWidth(150);
        b.setHeight(150);
        b.setText("click");
        l.addView(b);

        FrameLayout f = new FrameLayout(this);
        f.setLayoutParams(new FrameLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT));
        f.setBackgroundColor(getColor(android.R.color.holo_blue_light));
        l.addView(f);

    }

    public static int getDarkerColor(int color) {
        float[] hsv = new float[3];
        Color.colorToHSV(color, hsv);
        hsv[2] *= 0.9f;
        return Color.HSVToColor(hsv);
    }

    public static int getLighterColor(int color) {
        float[] hsv = new float[3];
        Color.colorToHSV(color, hsv);
        hsv[2] = 1f;
        return Color.HSVToColor(hsv);
    }
}

File: Android/Android Animations/AnimatingDrawables/app/src/main/java/com/wparja/myapplication/MainActivity.kt
package com.wparja.myapplication

import android.graphics.drawable.AnimatedVectorDrawable
import android.graphics.drawable.AnimationDrawable
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private lateinit var batteryAnimation: AnimationDrawable

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var isChecked = true

        animatedVectorDrawable.setOnClickListener {
            if (isChecked) {
                checkToClose()
            } else {
                closeToClose()
            }
            isChecked = !isChecked
        }

    }

    private fun closeToClose() {
        animatedVectorDrawable.setImageResource(R.drawable.avd_reverse_anim)
        val avdCloseToCheck = animatedVectorDrawable.drawable as AnimatedVectorDrawable
        avdCloseToCheck.start()
    }

    private fun checkToClose() {
        animatedVectorDrawable.setImageResource(R.drawable.avd_anim)
        val avdCheckToClose = animatedVectorDrawable.drawable as AnimatedVectorDrawable
        avdCheckToClose.start()
    }

    override fun onStart() {
        super.onStart()
        imageView.setBackgroundResource(R.drawable.battery_animation_list)
        batteryAnimation = imageView.background as AnimationDrawable
        batteryAnimation.start()
    }

    fun startAnimation(view: View) {
        if (batteryAnimation.isRunning) {
            batteryAnimation.stop()
        } else {
            batteryAnimation.start()
        }
    }
}